site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md

```diff
@@ -0,0 +1,261 @@
+
+## Overview of steps
+
+* Download and extract Velero v1.1
+* Deploy and Configure a Minio Object store
+* Install Velero using the `velero install` command, ensuring that both `restic` support and a Minio `publicUrl` are included
+* Run a test backup/restore of a stateless application that has been deployed on upstream Kubernetes
+
+## What this post does not show
+
+* How to install Velero v1.1 with PKS, Pivotal Container Service, on vSphere
+* A demonstration on how to do backup/restore of a stateful application (i.e. PVs)
+* The assumption is that the Kubernetes nodes in your cluster have internet access in order to pull the Velero images. This guide does not show how to add images using a local repository
+
+## Download and extract Velero v1.1
+
+The [Velero v1.1 binary can be found here](https://github.com/heptio/velero/releases/tag/v1.1.0.). Download and extract it to the desktop where you wish to manage your Velero backups, then copy or move the `velero` binary to somewhere in your $PATH.
+
+## Deploy and Configure a Minio Object Store as a backup destination
+
+Velero sends data and metadata about the Kubernetes objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backsup can be kept on-premises.
+
+* Note: Stateful backups of applications deployed in Kubernetes on vSphere that use the `restic` plugin for backing up Persistent Volumes send the backup data to the same S3 Object Store.
+
+There are a few different steps required to successfully deploy the Minio S3 Object Store.
+
+### 1. Create a Minio credentials secret file
+
+A simple credentials file containing the login/password (id/key) for the local on-premises Minio S3 Object Store must be created.
+
+```bash
+$ cat credentials-velero
+[default]
+aws_access_key_id = minio
+aws_secret_access_key = minio123
+```
+
+### 2. Expose Minio Service on a NodePort
+
+While this step is optional, it is useful for two reasons. The first is that it gives you a way to access the Minio portal through a browser and examine the backups. The second is that it enables you to specify a `publicUrl` for Minio, which in turn means that you can access backup and restore logs from the Minio S3 Object Store.
+
+To expose the Minio Service on a NodePort, a modification of the `examples/minio/00-minio-deployment.yaml` manifest is necessary. The only change is to the type: field, from ClusterIP to NodePort:
```

Comment:
See my comment on the other post about editing the deployment.

Reply:
Yep - @skriss  confirmed this is how to do it. Also worked this way in my testing.